.github/workflows/linux-build-and-test.yaml

```diff
@@ -0,0 +1,143 @@
+name: Regression Linux
+on:
+  schedule:
+    # run daily 20:00 on master branch
+    - cron: '0 20 * * *'
+  push:
+    branches:
+      - prerelease_test
+  pull_request:
+jobs:
+  regress_linux:
+    name: PG${{ matrix.pg }} [${{ matrix.type }}]
+    runs-on: ubuntu-18.04
```

Comment:
I had hoped that this would cover also Mac. My original config for this (which I believe you were going to base this off of) supported Mac. This seems like a regression and not the improvement we are hoping for over Travis.

Reply:
i think mac should be a separate workflow, otherwise we need platform specific steps to support certain features like getting core dumps, having it separate would also prevent bloating the matrix further

Reply:
This would lead to a lot of duplication of this configuration because there _should_ essentially be no difference between running these steps on a Mac vs Linux. I actually spent a fair amount of time in making my original config work across both OSes, so I think it would be a shame to abandon multi-OS support now. Maybe we should have further discussion about this? 

Core dumps is a whole different issue, and we don't support it right now. I'd rather split into two workflows only when necessary.

Reply:
FWIW, this was also one of the reasons I built PG from source as it makes the steps OS agnostic compared to custom install steps.